src/test/java/dev/jbang/cli/TestRun.java

```diff
@@ -1250,6 +1250,30 @@ void multiSourcesHttp() throws IOException {
 
 	}
 
+	String script = "" +
+			"class One {\n" +
+			"}" +
+			"class Two {\n" +
+			"public static void main(String... args) { };\n" +
```

Comment:
when I remove this line the test still passes so its not actually testing the issue.

Reply:
i.e. building is not the issue running is.

Reply:
actually it is build that is the issue but code and test doesn't catch it . i'll update the pr